board/embeddedartists/mx7ulpea-ucom/mx7ulpea-ucom.c

```diff
@@ -87,6 +106,264 @@ int board_early_init_f(void)
 	return 0;
 }
 
+#if defined(CONFIG_OPTEE) && defined(CONFIG_FPGA)
```

Comment:
Probably a good ideal to allow having the update/flash/boot logic still work without OPTEE, as that way we can test it independently. 

You can protect just the TEE stuff (compare hash, etc) with CONFIG_OPTEE, that way we can still test this without OPTEE (which is useful during bringup).